### PR TITLE
chore: add segment tracking (PLATFORM-2626)

### PR DESCRIPTION
**Pull request by johnsca**
Add Segment tracking analytics code to Redash HTML.

Implements [PLATFORM-2626](https://stacklet.atlassian.net/browse/PLATFORM-2626)
Depends on https://github.com/stacklet/platform-ui/pull/1194

### CODE REVIEW

**Comment by johnsca** [comment]
This is deployed to [my sandbox](https://console.cory.sandbox.stacklet.dev/) for testing.

**Comment by tvansteenburgh** [comment]
Hmm, the Segment key would preferably not be hard-coded, since we use a different one for internal vs customer envs. The Segment key is stored with the feature flags, and we use a TF override to set it differently for internal envs, for example: https://github.com/stacklet/imp-manifests/blob/main/qa/shared-infra/application-config/settings.auto.tfvars

**Review by fwereade — APPROVED** [review]
LGTM, thank you

**Comment by johnsca** [comment]
Ok, key is now read from cookie and CSP issues have been resolved.

File: redash/settings/__init__.py [hunk 1]
```diff
@@ -142,7 +142,7 @@
 # for more information. E.g.:
 CONTENT_SECURITY_POLICY = os.environ.get(
     "REDASH_CONTENT_SECURITY_POLICY",
-    "default-src 'self'; style-src 'self' 'unsafe-inline'; script-src 'self' 'unsafe-eval'; font-src 'self' data:; img-src 'self' http: https: data: blob:; object-src 'none'; frame-ancestors 'none'; frame-src redash.io;",
+    "default-src 'self'; style-src 'self' 'unsafe-inline'; script-src 'self' 'unsafe-eval' *.segment.com *.segment.io; connect-src 'self' *.segment.com *.segment.io; font-src 'self' data:; img-src 'self' http: https: data: blob:; object-src 'none'; frame-ancestors 'none'; frame-src redash.io;",
```
Comment (johnsca) [inline redash/settings/__init__.py]: I'm still seeing CSP errors for Hotjar, but I don't actually see where that's coming from in this repo. Should I add `*.hotjar.com` to this header?

Comment (tvansteenburgh) [reply redash/settings/__init__.py]: Yes. It's because the Segment-to-Hotjar integration is client-side.

**Pull request merged**